tasks/main.yml

```diff
@@ -0,0 +1,55 @@
+---
+- name: Set version specific variables
+  include_vars: "{{ item }}"
+  with_first_found:
+    - "{{ ansible_distribution }}-{{ ansible_distribution_version }}.yml"
+    - "{{ ansible_distribution }}-{{ ansible_distribution_major_version }}.yml"
+    - "{{ ansible_distribution }}.yml"
+    - "{{ ansible_os_family }}.yml"
+    - "default.yml"
+
+- name: Ensure certificate role dependencies are installed
+  package:
+    name: "{{ __certificate_packages }}"
+    state: present
+
+- name: Find required providers
+  set_fact:
```

Comment:
As this variable is used in only a few tasks below, please wrap those tasks in a `block:` and set the variable on a block, instead of making it a fact. (facts are globals and can not be unset, so a local variable is preferred if applicable.)

Reply:
20269498b23980fda8691698e476bbedac86d65f